### PR TITLE
Add the workflow for AWS custom security groups feature

### PR DESCRIPTION
**Pull request by gpei**
For the CI task in https://issues.redhat.com/browse/CORS-2639

### CODE REVIEW

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-aws

**Comment by gpei** [comment]
@yunjiang29 @jianlinliu please have a check when you have time, thanks

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-aws

**Comment by yunjiang29** [comment]
Test
```
controlPlane:
  platform:
    aws:
      zones:
      - "us-east-1a"
      - "us-east-1b"
      - "us-east-1c"
      additionalSecurityGroupIDs:
      - "sg-02ef3092e4b29333f"
      - "sg-08aceef2aac55ca6c"
      - "sg-01694a774c2281303"
      type: m6i.xlarge
  architecture: amd64
  name: master
compute:
- platform:
    aws:
      zones:
      - "us-east-1a"
      - "us-east-1b"
      - "us-east-1c"
      additionalSecurityGroupIDs:
      - "sg-02ef3092e4b29333f"
      - "sg-08aceef2aac55ca6c"
      - "sg-01694a774c2281303"
      type: m5.xlarge
```

File: ci-operator/step-registry/ipi/conf/aws/custom-security-groups/ipi-conf-aws-custom-security-groups-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    name: "4.12"
+    tag: upi-installer
+  grace_period: 10m
+  commands: ipi-conf-aws-custom-security-groups-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  env:
+  - name: ENABLE_SHARED_PHZ
```
Comment (yunjiang29) [inline ci-operator/step-registry/ipi/conf/aws/custom-security-groups/ipi-conf-aws-custom-security-groups-ref.yaml:14]: @gpei `ENABLE_SHARED_PHZ` is not required for this step, we can remove it.

Comment (gpei) [reply ci-operator/step-registry/ipi/conf/aws/custom-security-groups/ipi-conf-aws-custom-security-groups-ref.yaml:14]: ok, let me do it

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 2]
```diff
@@ -267,8 +267,8 @@ tests:
       TAG_VERSION: '@4.14'
     test:
     - chain: openshift-e2e-test-qe
-    workflow: cucushift-installer-rehearse-aws-ipi
-- as: aws-ipi-compact-p3-f28-destructive
+    workflow: cucushift-installer-rehearse-aws-ipi-custom-security-groups
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:270]: We are also testing worker customer sg, sounds like a compact cluster is not very proper for the testing?

Comment (gpei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:270]: Ah, good point, let me choose another one to combine

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-baselinecaps-none-custom-sg-p2-f14

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-baselinecaps-none-custom-sg-p2-f14

**Comment by gpei** [comment]
Since lxia is PTO, @jhou1 @kuiwang02 could you help to approve this?

**Comment by kuiwang02** [comment]
@gpei i am in approver list from golang repo's view, not regression test's view.

**Comment by gpei** [comment]
> @gpei i am in approver list from golang repo's view, not regression test's view.

Sorry for the noise, got your name from the following files mentioned by the ci-bot.
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-tests-private/OWNERS
https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/openshift-tests-private/OWNERS

**Comment by kuiwang02** [comment]
> > @gpei i am in approver list from golang repo's view, not regression test's view.
> 
> Sorry for the noise, got your name from the following files mentioned by the ci-bot. https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-tests-private/OWNERS https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/openshift-tests-private/OWNERS

yes. I am in approver list to approve tests-private repo related stuff, not regression test stuff.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 3]
```diff
@@ -128,7 +128,7 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-aws-c2s-ipi-disconnected-private
-- as: aws-ipi-baselinecaps-none-p2-f14
+- as: aws-ipi-baselinecaps-none-custom-sg-p2-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:131]: btw, can we add the similar testing in arm ci jobs.

**Pull request merged**